src/c_api.cpp

```diff
@@ -113,6 +114,8 @@ class Booster {
 
   Booster(const Dataset* train_data,
           const char* parameters) {
+    int i = INT_MIN;
+    int j = -i;
```

Comment:
it seems that the sanitizers test is passing. I'm adding this temporarily to try to trigger a failure. I found this example of something that will trigger a runtime error from this sanitizer [here](https://developers.redhat.com/blog/2014/10/16/gcc-undefined-behavior-sanitizer-ubsan/)

Reply:
well that one did not trigger any issues. I'll try something else tonight